File: dev/20190822/k2works/src/Domain/Type/FizzBuzzType03.ts
import { IFizzBuzzType } from "../../Application/Service/IFizzBuzzType";
import { FizzBuzzValue } from "../Model/FizzBuzzValue";

export class FizzBuzzType03 extends IFizzBuzzType {
  public generate(input: number): FizzBuzzValue {
    const isFizz: boolean = input % 3 === 0;
    const isBuzz: boolean = input % 5 === 0;

    if (isFizz && isBuzz) {
      return new FizzBuzzValue(input, IFizzBuzzType.FIZZ_BUZZ);
    }
    return new FizzBuzzValue(input, input.toString());
  }
}

File: dev/20190902/k2works/app/FizzBuzz/Domain/Type/FizzBuzzType03.cs
using FizzBuzz.Application.Service;
using FizzBuzz.Domain.Model;

namespace FizzBuzz.Domain.Type
{
    public class FizzBuzzType03: FizzBuzzType
    {
        public override IFizzBuzz Generate(int number)
        {
            var isFizz = number % 3 == 0;
            var isBuzz = number % 5 == 0;
            if (isFizz && isBuzz)
            {
                return new FizzBuzzValue(number,FIZZ_BUZZ);
            }

            return new FizzBuzzValue(number,number.ToString());
        }
    }
}